### PR TITLE
Fix lexing of integer literals with leading zero

### PR DESCRIPTION
**Pull request by g4rgoil**

### CODE REVIEW

File: run [hunk 1]
```diff
@@ -17,4 +17,4 @@ EXE_PATH=$(find $SCRIPT_PATH/target/java-compiler-*-with-dependencies.jar) || ex
 
 # execute the compiler with the provided arguments and return the exit code
 java -jar "$EXE_PATH" "$@"
-exit $?
+exit $?
```
Comment (larsk21) [inline run:20]: Is this part of fixing the lexing of integer literals?

File: src/test/java/edu/kit/compiler/lexer/LexerTest.java [hunk 2]
```diff
@@ -48,6 +48,11 @@ public void testUnicode() throws LexException {
         var lexer = new Lexer(getIterator("int \u309e = 1;"));
         assertEquals(Keyword_Int, lexer.getNextToken().getType());
         assertThrows(LexException.class, () -> lexer.getNextToken());
+        var lexer2 = new Lexer(getIterator("int 😀 = 1;"));
```
Comment (larsk21) [inline src/test/java/edu/kit/compiler/lexer/LexerTest.java:51]: Same here, shouldn't this be in its own PR?

File: src/test/java/edu/kit/compiler/lexer/LexerTest.java [hunk 3]
```diff
@@ -168,7 +173,8 @@ public void testZeroLiteral() throws LexException {
     @Test
     public void testIllegalLiteral() throws LexException {
```
Comment (larsk21) [inline src/test/java/edu/kit/compiler/lexer/LexerTest.java:174]: This is not illegal anymore, please rename this test.

**Review by larsk21 — CHANGES_REQUESTED** [review]

File: run (hunk 1 above)
Comment (g4rgoil) [reply run:20]: No, it's an unrelated commit I hadn't noticed.

File: src/test/java/edu/kit/compiler/lexer/LexerTest.java (hunk 2 above)
Comment (g4rgoil) [reply src/test/java/edu/kit/compiler/lexer/LexerTest.java:51]: This one didn't make it into the original lexer PR. It's technically unrelated to this fix.

File: src/test/java/edu/kit/compiler/lexer/LexerTest.java [hunk 4]
```diff
@@ -166,9 +171,10 @@ public void testZeroLiteral() throws LexException {
     }
 
     @Test
-    public void testIllegalLiteral() throws LexException {
+    public void testLeadingZero() throws LexException {
```
Comment (larsk21) [inline src/test/java/edu/kit/compiler/lexer/LexerTest.java:174]: Just one last thing, could you also remove the `throws` declaration, now that there's nothing that should be thrown anymore? 😅

Comment (g4rgoil) [reply src/test/java/edu/kit/compiler/lexer/LexerTest.java:174]: For the test to compile, it still needs a `throws` declaration, because we're using checked exceptions in the lexer. The test would still fail if an exception we're to be thrown.

**Review by larsk21 — APPROVED** [review]
